src/testRunner/unittests/tsbuild/moduleSpecifiers.ts

```diff
@@ -0,0 +1,59 @@
+                "/src/common/nominal.ts": "export declare type Nominal<T, Name extends string> = T & {\n\t[Symbol.species]: Name;\n};\n",
+                "/src/common/tsconfig.json": JSON.stringify({
+                    extends: "../../tsconfig.base.json",
+                    compilerOptions: { composite: true },
+                    include: ["nominal.ts"]
+                }),
+                "/src/sub-project/index.ts": "import { Nominal } from '../common/nominal';\n\nexport type MyNominal = Nominal<string, 'MyNominal'>;\n",
+                "/src/sub-project/tsconfig.json": JSON.stringify({
+                    extends: "../../tsconfig.base.json",
+                    compilerOptions: { composite: true },
+                    references: [{ path: "../common" }],
+                    include: ["./index.ts"]
+                }),
+                "/src/sub-project-2/index.ts":
+                    "import { MyNominal } from '../sub-project/index';\n\n" +
+                    "const variable = {\n\tkey: 'value' as MyNominal,\n};\n\n" +
+                    "export function getVar(): keyof typeof variable {\n\treturn 'key';\n}\n",
+                "/src/sub-project-2/tsconfig.json": JSON.stringify({
+                    extends: "../../tsconfig.base.json",
+                    compilerOptions: { composite: true },
+                    references: [{ path: "../sub-project" }],
+                    include: ["./index.ts"]
+                }),
+                "/src/tsconfig.json": JSON.stringify({
+                    compilerOptions: { composite: true },
+                    references: [{ path: "./sub-project" }, { path: "./sub-project-2" }],
+                    include: []
+                }),
+                "/tsconfig.base.json": JSON.stringify({
+                    compilerOptions: {
+                        rootDir: "./",
+                        outDir: "lib",
+                        lib: ["dom", "es2015", "es2015.symbol.wellknown"]
+                    },
+                }),
+                "/tsconfig.json": JSON.stringify({
+                    compilerOptions: { composite: true },
+                    references: [{ path: "./src" }],
+                    include: []
+                })
+            },
+            cwd: "/"
+        });
+        const sys = new fakes.System(fs, { executingFilePath: "/", newLine: "\n" });
+        const host = new fakes.SolutionBuilderHost(sys);
+        const builder = createSolutionBuilder(host, ["/tsconfig.json"], { dry: false, force: false, verbose: false });
+        builder.build("/tsconfig.json");
+
+        const data = host.vfs.readFileSync("/lib/src/sub-project-2/index.d.ts", "utf8");
+
+        // Prior to fixing GH31696 the import below was `import("../../lib/src/common/nonterminal")`
+        assert.equal(data, `declare const variable: {\n    key: import("../common/nominal").Nominal<string, "MyNominal">;\n};\nexport declare function getVar(): keyof typeof variable;\nexport {};\n`);

```

Comment:
Instead of this can you baseline the change. Try `verifyTsbuildOutput` You can pass `baselineonly` option

Reply:
I wanted to keep this simple. `verifyTsbuildOutput` seemed to do quite a bit more than was strictly necessary to test this.

Reply:
It also emits quite a bit of excess information unrelated to what I'm actually trying to verify, since it emits the entire file system diff.

Reply:
These kind of verification do not perform well when we change our emit. So baselines are better

Reply:
I've switched to baselines, but still feel that `verifyTsbuildOutput` was too complex for what I needed to verify.

Reply:
I disagree. With just `baselineOnly` option it just baselines the build. You can specify only initial build and it will baseline only that in a single file. Its like our compiler test runner that has complexities but does baseline outputs well. Having two types of baselines is just confusing and error prone later if we say want to add what happens in incremental scenario or something like that.

Reply:
I looked into migrating it to `verifyTsbuildOutput` and I'm not convinced. The setup to use that function seems unnecessarily complicated and is undocumented. I'd much rather keep the test simple and switch to `verifyTsbuildOutput` if we ever need any of that added complexity. I can switch to emitting the single file patch using the same path format other tsbuild tests use, but I'd rather keep the test as focused as possible for the time being.